### PR TITLE
[13.0] [MIG] account_invoice_triple_discount: Migration to 13.0

### PR DESCRIPTION
**Pull request by CarlosRoca13**
cc @Tecnativa TT31757

This PR is for cherry pick the commit done by @benwillig for the migration and @dreispt for the fix when the module was migrated to v14.0.

@pedrobaeza @ernestotejeda please review it

### CODE REVIEW

**Comment by CarlosRoca13** [comment]
I will apply the corrections discussed in this PR https://github.com/OCA/account-invoicing/pull/733 to resolve all discussions.

**Review by dreispt — APPROVED** [review]

**Comment by CarlosRoca13** [comment]
This PR superseds this other: https://github.com/OCA/account-invoicing/pull/733

Ping @chienandalu

File: account_invoice_triple_discount/models/account_move.py [hunk 1]
```diff
@@ -0,0 +1,128 @@
...
+            values = old_values[index]
+            if values:
+                record.write(old_values[index])
+        return records
+
+    @api.onchange(
+        "discount", "price_unit", "tax_ids", "quantity", "discount2", "discount3",
+    )
+    def _onchange_price_subtotal(self):
+        return super(AccountMoveLine, self)._onchange_price_subtotal()
```
Comment (ernestotejeda) [inline account_invoice_triple_discount/models/account_move.py]:
```suggestion
        return super()._onchange_price_subtotal()
```

File: account_invoice_triple_discount/models/account_move.py [hunk 2]
```diff
@@ -0,0 +1,128 @@
...
+
+    def _get_price_total_and_subtotal(self, **kwargs):
+        self.ensure_one()
+        kwargs["discount"] = self._compute_aggregated_discount(
+            kwargs.get("discount") or self.discount
+        )
+        return super(AccountMoveLine, self)._get_price_total_and_subtotal(**kwargs)
```
Comment (ernestotejeda) [inline account_invoice_triple_discount/models/account_move.py]:
```suggestion
        return super()._get_price_total_and_subtotal(**kwargs)
```

File: account_invoice_triple_discount/models/account_move.py [hunk 3]
```diff
@@ -0,0 +1,128 @@
...
+
+    def _get_fields_onchange_balance(self, **kwargs):
+        self.ensure_one()
+        kwargs["discount"] = self._compute_aggregated_discount(
+            kwargs.get("discount") or self.discount
+        )
+        return super(AccountMoveLine, self)._get_fields_onchange_balance(**kwargs)
```
Comment (ernestotejeda) [inline account_invoice_triple_discount/models/account_move.py]:
```suggestion
        return super()._get_fields_onchange_balance(**kwargs)
```

**Review by ernestotejeda — COMMENTED** [review]
Functional review: 👍🏼

**Review by ernestotejeda — APPROVED** [review]

**Review by chienandalu — APPROVED** [review]

**Pull request merged**